pkg/upgrade/control_plane.go

```diff
@@ -529,6 +531,10 @@ func (u *ControlPlaneUpgrader) updateMachine(replacementKey ctrlclient.ObjectKey
 		return err
 	}
 
+	if err := u.updateSecrets(bootstrap); err != nil {
+		return err
+	}
+
```

Comment:
Hmm...I'm guessing we want to do this before the original machine is removed in order to prevent the garbage collection of the original secrets. If that's the case then maybe a comment here describing that.

Reply:
Right, we can only shift them over between the time where the replacement becomes healthy and before we remove the old, else they might be GC'd. Dropped a comment there. Thanks!